test/unit/specs/components/Personalization/createRedirectHandler.spec.js

```diff
@@ -29,21 +30,41 @@ describe("Personalization::createRedirectDecisionHandler", () => {
   };
   const replace = jasmine.createSpy();
 
+  const window = {
+    location: { replace }
+  };
+
   beforeEach(() => {
     collect = jasmine.createSpy().and.returnValue(Promise.resolve());
+    logger = jasmine.createSpyObj("logger", ["warn"]);
+    showContainers = jasmine.createSpy("showContainers");
   });
 
   it("should trigger collect before redirect", () => {
-    const win = {
-      location: { replace }
-    };
     const handleRedirectDecisions = createRedirectHandler({
       collect,
-      win
+      window,
+      logger,
+      showContainers
     });
     return handleRedirectDecisions(decisions).then(() => {
-      expect(collect).toHaveBeenCalledWith({ meta }, documentMayUnload);
+      expect(collect).toHaveBeenCalledWith({ meta, documentMayUnload });
       expect(replace).toHaveBeenCalledWith(decisions[0].items[0].data.content);
     });
   });
+  it("should trigger showContainers and logger at redirect", () => {
```

Comment:
This description should probably mention something about an error being thrown when attempting to redirect.